### ISSUE TITLE
generated systemd units lack values of environment variables referenced

### ISSUE THREAD

**Issue by thmo** [containers/podman#10101]
/kind bug

**Description**

When generating systemd units using `--new`, the value of environment variables which were only referenced in the original `create` command is lost.

**Steps to reproduce the issue:**

```shell
$ export FOO=bar
$ podman create --name=test --env FOO alpine
$ podman inspect test | grep bar
$ podman generate systemd -n --new test | grep bar
```

**Describe the results you received:**

"bar" is found for the `inspect` command, but not found in the systemd unit.

**Describe the results you expected:**

I expected the value of `FOO` being saved somewhere, either in the systemd unit itself, or a in separate file referenced by the systemd unit.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
$ podman version 
Version:      3.1.0-dev
API Version:  3.1.0-dev
Go Version:   go1.16.1
Built:        Fri Mar 26 19:32:03 2021
OS/Arch:      linux/amd64
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.1.0-0.13.module_el8.5.0+733+9bb5dffa.x86_64
```

**Comment by mheon** [comment]
Hm. @vrothberg This one looks tricky.

Should we parse the CLI to look for bare (key, no value) `--env` flags and look them up / include them in the generated unit file?

**Comment by thmo** [comment]
Wouldn't it be better to record the names of those variables upon container creation time and store it somewhere inside the container information structure, in addition to the the `Config.CreateCommand` entry? Parsing that entry seems a bit dangerous.

**Comment by vrothberg** [comment]
> Should we parse the CLI to look for bare (key, no value) `--env` flags and look them up / include them in the generated unit file?

Yes.  We're reparsing the CLI args already for _some_ flags already when generating the unit files. I haven't had a look at the env code but I _think_ that env is evaluated at container-creation time. Maybe we can just replace the raw `--env` with the evaluated ones.

**Comment by mheon** [comment]
I did a quick verification, and you're correct. We are evaluating at container creation time, so the full environment variable key-value pairs are available in the container's OCI spec.

**Comment by Luap99** [comment]
PR #10236 should fix this.
